Sources/DatadogSDKTesting/DDTestSession.swift

```diff
@@ -0,0 +1,244 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2020-2021 Datadog, Inc.
+ */
+
+import Foundation
+@_implementationOnly import OpenTelemetryApi
+@_implementationOnly import OpenTelemetrySdk
+@_implementationOnly import SigmaSwiftStatistics
+
+public class DDTestSession: NSObject {
+    var bundleName = ""
+    var bundleFunctionInfo = FunctionMap()
+    var codeOwners: CodeOwners?
+    var testFramework = "Swift API"
+
+    public init(bundleName: String) {
+        if DDTestMonitor.instance == nil {
+            DDTestMonitor.installTestMonitor()
+        }
+
+        self.bundleName = bundleName
+#if !os(tvOS) && (targetEnvironment(simulator) || os(macOS))
+        DDSymbolicator.createDSYMFileIfNeeded(forImageName: bundleName)
+        bundleFunctionInfo = FileLocator.testFunctionsInModule(bundleName)
+#endif
+        if let workspacePath = DDTestMonitor.env.workspacePath {
+            codeOwners = CodeOwners(workspacePath: URL(fileURLWithPath: workspacePath))
+        }
+
+        if !DDTestMonitor.env.disableCrashHandler {
+            DDCrashes.install()
+        }
+    }
+
+    public func end() {
+        /// We need to wait for all the traces to be written to the backend before exiting
+        DDTestMonitor.tracer.flush()
+    }
+}
+
+@objc public enum DDTestStatus: Int {
+    case pass
+    case fail
+    case skip
+}
+
+public extension DDTestSession {
+    // Public interface for DDTestSession
+
+    @objc func suiteStart(name: String) -> DDTestSuite {
+        let suite = DDTestSuite(name: name)
+        return suite
+    }
+
+    @objc(suiteEnd:) func suiteEnd(suite: DDTestSuite) {
+        suite.end()
+    }
+
+    @objc func testStart(name: String, suite: DDTestSuite) -> DDTest {
+        return DDTest(name: name, suite: suite, session: self)
+    }
+
+    @objc(testSetAttribute:key:value:) func testSetAttribute(test: DDTest, key: String, value: Any) {
+        test.setAttribute(key: key, value: value)
+    }
+
+    @objc(testSetErrorInfo:type:message:callstack:) func testSetErrorInfo(test: DDTest, type: String, message: String, callstack: String?) {
+        test.setErrorInfo(type: type, message: message, callstack: callstack)
+    }
+
+    @objc(testAddBenchmark:name:samples:info:) func testAddBenchmark(test:DDTest, name: String, samples: [Double], info: String?) {
+        test.addBenchmark(name: name, samples: samples, info: info)
+    }
+
+    @objc(testEnd:status:) func testEnd(test: DDTest, status: DDTestStatus) {
+        test.end(status: status)
+    }
+
+}
+
+public class DDTestSuite: NSObject {
```

Comment:
As we're using it from `@objc` API:
```swift
@objc func testStart(name: String, suite: DDTestSuite) -> DDTest
```
I think this also requires `@objc` annotation, no?

To be ultimately sure on this, we might consider adding smoke test for Objective-C interface in `dd-swift-testing` as we do [here for `dd-sdk-ios`](https://github.com/DataDog/dd-sdk-ios/blob/82dd2ef2bea5705ddb7d73e16e6ab9d44d6d3531/Tests/DatadogTests/DatadogObjc/ObjcAPITests/DDRUMMonitor%2BapiTests.m#L51) (just adding `.m` test with calling the API to ensure they are visible in Objective-C).

Reply:
I would say that classes don't need the annotation if they inherit from a ObjC class, they are automatically made visible, but as you say is better having a test for assuring it.